pydicom/tests/test_values.py

```diff
@@ -0,0 +1,153 @@
+        forget to escape any regex special characters like '|', '(', ')', etc.
+    func : callable
+        The function that is expected to raise the exception.
+    args
+        The callable function `func`'s arguments.
+    kwargs
+        The callable function `func`'s keyword arguments.
+
+    Notes
+    -----
+    Taken from https://github.com/glemaitre/specio, BSD 3 license.
+    """
+    with pytest.raises(type_error) as excinfo:
+        func(*args, **kwargs)
+    excinfo.match(message)
+
+
+class TestConvertTag(object):
+    def test_big_endian(self):
+        """Test convert_tag with a big endian byte string"""
+        bytestring = b'\x00\x10\x00\x20'
+        assert convert_tag(bytestring, False) == Tag(0x0010, 0x0020)
+
+    def test_little_endian(self):
+        """Test convert_tag with a little endian byte string"""
+        bytestring = b'\x10\x00\x20\x00'
+        assert convert_tag(bytestring, True) == Tag(0x0010, 0x0020)
+
+    def test_offset(self):
+        """Test convert_tag with an offset"""
+        bytestring = b'\x12\x23\x10\x00\x20\x00\x34\x45'
+        assert convert_tag(bytestring, True, 0) == Tag(0x2312, 0x0010)
+        assert convert_tag(bytestring, True, 2) == Tag(0x0010, 0x0020)
+
+    @pytest.mark.skip(reason='empty bytestring not handled properly')
+    def test_empty_bytestring(self):
+        """Test convert_tag with empty bytestring"""
+        bytestring = b''
+        assert convert_tag(bytestring, True) == ''
+
+    @pytest.mark.skip(reason='bad bytestring not handled properly')
+    def test_bad_bytestring(self):
+        """Test convert_tag with a bad bytestring"""
+        bytestring = b'\x10\x00'
+        convert_tag(bytestring, True)
+
+
+class TestConvertAT(object):
+    def test_big_endian(self):
+        """Test convert_ATvalue with a big endian byte string"""
+        # VM 1
+        bytestring = b'\x00\x10\x00\x20'
+        assert convert_ATvalue(bytestring, False) == Tag(0x0010, 0x0020)
+
+        # VM 3
+        bytestring += b'\x00\x10\x00\x30\x00\x10\x00\x40'
+        out = convert_ATvalue(bytestring, False)
+        assert Tag(0x0010, 0x0020) in out
+        assert Tag(0x0010, 0x0030) in out
+        assert Tag(0x0010, 0x0040) in out
+
+    def test_little_endian(self):
+        """Test convert_ATvalue with a little endian byte string"""
+        # VM 1
+        bytestring = b'\x10\x00\x20\x00'
+        assert convert_ATvalue(bytestring, True) == Tag(0x0010, 0x0020)
+
+        # VM 3
+        bytestring += b'\x10\x00\x30\x00\x10\x00\x40\x00'
+        out = convert_ATvalue(bytestring, True)
+        assert Tag(0x0010, 0x0020) in out
+        assert Tag(0x0010, 0x0030) in out
+        assert Tag(0x0010, 0x0040) in out
+
+    def test_empty_bytestring(self):
+        """Test convert_ATvalue with empty bytestring"""
+        bytestring = b''
+        assert convert_ATvalue(bytestring, True) == []
+
+    @pytest.mark.skip(reason='bad bytestring not handled properly')
```

Comment:
skip?

Reply:
Just an issue for that would be sufficient. Also, @scaramallion already added a respective TODO in the code, so I guess he will do that next.

Reply:
I understand. But I would still feel more comfortable if we mark them for instance `@pytest.mark.todo` and add `-m 'not todo'` in the CI. Its easy to `run pytest -m 'todo' ` locally to make sure that everything is fine. 

**Even if all this is temporarly and `-m 'not todo'` should be removed from the CI ASAP**

Reply:
That sounds ok to me. I also would prefer this approach for unit tests that are not implemented yet (see above), instead of throwing an exception (which would fail the build unnecessary) or just add a pass (which would make it easy to forget them).

Reply:
> maybe we can add a label like `ci-ignore`, `things-we-know-and-we-are-working-on` or something.

I guess that I should had thought about `todo`. Because this is exactly what I was proposing. Then using `pass` or the error. I prefer the error. Because tests are meant to break. But this is a matter of taste. So I'll stop being picky.